app/oracle.py

```diff
@@ -167,50 +166,33 @@ def prompt(prompt_message, prompt_end):
 
 logging.info('Starting the main loop')
 while True:
-
-    # Get the frame and validators keys from ETH1 side
+    # Get the the epoch that is both finalized and reportable
     current_frame = oracle.functions.getCurrentFrame().call()
-    reportable_epoch = current_frame[0]
-    logging.info(f'Reportable epoch: {reportable_epoch}')
+    potentially_reportable_epoch = current_frame[0]
+    logging.info(f'Potentially reportable epoch: {potentially_reportable_epoch} (from ETH1 contract)')
+    finalized_epoch_beacon = beacon.get_finalized_epoch()
+    logging.info(f'Last finalized epoch: {finalized_epoch_beacon} (from Beacon)')
+    reportable_epoch = min(finalized_epoch_beacon, potentially_reportable_epoch) // epochs_per_frame * epochs_per_frame
+    reportable_slot = reportable_epoch * slots_per_epoch
+    logging.info(f'Reportable state: epoch:{reportable_epoch} slot:{reportable_slot}')
+
     validators_keys = get_validators_keys(registry)
     logging.info(f'Total validator keys in registry: {len(validators_keys)}')
-
-    # Wait for the epoch finalization on the beacon chain
-    while True:
-
-        finalized_epoch = beacon.get_finalized_epoch()  # take into account only finalized epoch
-        reportable_slot = reportable_epoch * slots_per_epoch  # it's possible that in epoch there are less slots
-
-        if reportable_epoch > finalized_epoch:
-            # The reportable epoch received from the contract
-            # is not finalized on the beacon chain so we are waiting
-            logging.info(
-                f'Reportable epoch: {reportable_epoch} is not finalized on Beacon. Finalized: {finalized_epoch}. Wait {await_time_in_sec} s'
-            )
-            time.sleep(await_time_in_sec)
-            continue
-        else:
-            logging.info(f'Reportable epoch ({reportable_epoch}) is finalized on beacon chain.')
-            break
-
-    # At this point the slot is finalized on the beacon
-    # so we are able to retrieve validators set and balances
     sum_balance, validators_on_beacon = beacon.get_balances(reportable_slot, validators_keys)
-    logging.info(f'ReportBeacon transaction arguments:')
-    logging.info(f'Reportable epoch: {reportable_epoch} Slot: {reportable_slot}')
-    logging.info(f'Sum balance in wei: {sum_balance}')
-    logging.info(f'Lido validators on Beacon chain: {validators_on_beacon}')
+
+    logging.info(f'Total balance on Beacon: {sum_balance} wei')
+    logging.info(f'Lido validators on Beacon: {validators_on_beacon}')
```

Comment:
these numbers alone are not enough to understand if the report looks normal-ish or not for an operator
let's add the time between the last reported epoch and currently reported, and profit/shashing calculations there, at a minimum